examples/src/multiple_render_targets/mod.rs

```diff
@@ -0,0 +1,547 @@
+        device: &wgpu::Device,
+        layout: &wgpu::BindGroupLayout,
+        texture_targets: &TextureTargets,
+        sampler: &wgpu::Sampler,
+    ) -> (wgpu::BindGroup, wgpu::BindGroup) {
+        let left = device.create_bind_group(&wgpu::BindGroupDescriptor {
+            layout,
+            entries: &[
+                wgpu::BindGroupEntry {
+                    binding: 0,
+                    resource: wgpu::BindingResource::TextureView(&texture_targets.a_view),
+                },
+                wgpu::BindGroupEntry {
+                    binding: 1,
+                    resource: wgpu::BindingResource::Sampler(sampler),
+                },
+            ],
+            label: None,
+        });
+
+        let right = device.create_bind_group(&wgpu::BindGroupDescriptor {
+            layout,
+            entries: &[
+                wgpu::BindGroupEntry {
+                    binding: 0,
+                    resource: wgpu::BindingResource::TextureView(&texture_targets.b_view),
+                },
+                wgpu::BindGroupEntry {
+                    binding: 1,
+                    resource: wgpu::BindingResource::Sampler(sampler),
+                },
+            ],
+            label: None,
+        });
+        (left, right)
+    }
+
+    fn draw(
+        &self,
+        encoder: &mut wgpu::CommandEncoder,
+        surface_view: &wgpu::TextureView,
+        width: u32,
+        height: u32,
+    ) {
+        let mut rpass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+            label: None,
+            color_attachments: &[Some(wgpu::RenderPassColorAttachment {
+                view: surface_view,
+                resolve_target: None,
+                ops: wgpu::Operations {
+                    load: wgpu::LoadOp::Clear(wgpu::Color::GREEN),
+                    store: wgpu::StoreOp::Store,
+                },
+            })],
+            depth_stencil_attachment: None,
+            timestamp_writes: None,
+            occlusion_query_set: None,
+        });
+        rpass.set_pipeline(&self.pipeline);
+        rpass.set_bind_group(0, &self.bindgroup_left, &[]);
+
+        let height = height as f32;
+        let half_w = width as f32 * 0.5;
+
+        // draw results in two separate viewports that split the screen:
+
+        rpass.set_viewport(0.0, 0.0, half_w, height, 0.0, 1.0);
+        rpass.draw(0..3, 0..1);
+
+        rpass.set_viewport(half_w, 0.0, half_w, height, 0.0, 1.0);
+        rpass.set_bind_group(0, &self.bindgroup_right, &[]);
+        rpass.draw(0..3, 0..1);
+    }
+
+    fn rebuild_resources(&mut self, device: &wgpu::Device, texture_targets: &TextureTargets) {
+        (self.bindgroup_left, self.bindgroup_right) = Self::create_bindgroups(
+            device,
+            &self.bindgroup_layout,
+            texture_targets,
+            &self.sampler,
+        )
+    }
+}
+
+struct TextureTargets {
+    a_view: wgpu::TextureView,
+    b_view: wgpu::TextureView,
```

Comment:
I think these could have better names. Maybe `red_target_view`, or something similar. Same change for the locals in TextureTargets::new